### PR TITLE
Ea/story31to33

### PR DESCRIPTION
**Pull request by eamouzou**
- Add test, revise controllers and views to make sure pets with approved applications cannot be deleted
- Add test, revise controllers and views to make sure deleting a pet removes it from favorites
- Add test, revise controllers and views to make sure flash Message for pet create and update
- Run migration to add default value for description

### CODE REVIEW

File: app/controllers/pets_controller.rb [hunk 1]
```diff
@@ -30,25 +36,37 @@ def update
...
   end
 
+
   def change_adoption
     pet = Pet.find(params[:petid])
     if pet.adoption_status == "adoptable"
       pet.update(adoption_status: "pending")
       pet_app = PetApplication.where(pet_id: params[:petid], application_id: params[:appid])[0]
       pet_app.update(adopted: true)
+      pet.update(approved: true)
```
Comment (eamouzou) [inline app/controllers/pets_controller.rb]: This confirms that the pet has been approved. There is an approved method in PetApplication but I wanted to be able to call this from pet.

File: app/controllers/pets_controller.rb [hunk 2]
```diff
@@ -30,25 +36,37 @@ def update
...
       approximate_age: params[:pet_approximate_age],
       sex: params[:pet_sex])
-    redirect_to "/pets/#{params[:id]}"
+      if pet.save
+        flash[:success] = "The pet has been updated."
+        redirect_to "/pets/#{params[:id]}"
+      else
+        flash[:error] = pet.errors.full_messages.to_sentence
+        redirect_to "/pets/#{pet.id}/edit"
+      end
```
Comment (eamouzou) [inline app/controllers/pets_controller.rb]: Same as above, flashes a message whether it saves or not.

File: app/models/pet.rb [hunk 3]
```diff
@@ -3,10 +3,14 @@ class Pet < ApplicationRecord
...
   before_save :default_values
+  validates_presence_of :name
+  validates_presence_of :approximate_age
+  validates_presence_of :sex
+  validates_presence_of :description
+
 
   def default_values
     self.adoption_status = "adoptable" if self.adoption_status.nil?
-    self.description = "Looking for a home" if self.description.nil?
```
Comment (eamouzou) [inline app/models/pet.rb]: This is removed because of the migration I ran. It is now a default in the table.

File: app/views/pets/index.html.erb [hunk 4]
```diff
@@ -6,6 +6,10 @@
   <h4>Sex: <%= pet.sex %></h4>
   <h4>Shelter Name: <a href="/shelters/<%= pet.shelter_id %>"><%= pet.shelter_name %></a></h4>
   <%=button_to "Edit #{pet.name}'s info!", "/pets/#{pet.id}/edit", :method => :get%>
-  <p><%=button_to "Delete #{pet.name}", "/pets/#{pet.id}", :method => :delete%></p>
+  <% if pet.approved == false %>
+    <p><%=button_to "Delete #{pet.name}", "/pets/#{pet.id}", :method => :delete%></p>
+  <% elsif pet.approved == true %>
+    <h4>You cannot delete this pet while its application is approved.</h4>
+  <% end %>
```
Comment (eamouzou) [inline app/views/pets/index.html.erb]: This conditional is added to remove the delete button if the pet has been approved.

File: db/migrate/20200210223302_add_default_to_pets_description.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddDefaultToPetsDescription < ActiveRecord::Migration[5.1]
+  def change
+    change_column :pets, :description, :string, :default => "Looking for a home"
+    Pet.all.where(description: nil).update_all(description: "Looking for a home")
+  end
+end
```
Comment (eamouzou) [inline db/migrate/20200210223302_add_default_to_pets_description.rb]: This is the migration for giving description a default value. It also goes through the records and updates this wherever that value is nil.

Comment (Will-Meighan) [reply db/migrate/20200210223302_add_default_to_pets_description.rb]: I like this addition. Very clean and is sensible.

**Comment by Will-Meighan** [comment]
Good work here. Cruising.

**Pull request merged**